File: atlas-sdk/JavaScript/event.js
const crypto = require("crypto");

class Event {
  constructor(type, source = "js-sdk", payload = null) {
    this.id = crypto.randomUUID();
    this.type = type;
    this.source = source;
    this.payload = payload || {};
    this.timestamp = Date.now();
    this.priority = "medium";
  }

  toDict() {
    return {
      id: this.id,
      type: this.type,
      source: this.source,
      payload: this.payload,
      timestamp: this.timestamp,
      priority: this.priority,
    };
  }

  static fromDict(data) {
    const e = new Event(data.type, data.source || "js-sdk", data.payload);
    e.id = data.id || e.id;
    e.timestamp = data.timestamp || Date.now();
    e.priority = data.priority || "medium";
    return e;
  }
}

module.exports = { Event };

File: atlas-sdk/Go/event.go
package atlas

import (
	"fmt"
	"time"
)

type Event struct {
	ID        string                 `json:"id"`
	Type      string                 `json:"type"`
	Source    string                 `json:"source"`
	Payload   map[string]interface{} `json:"payload"`
	Timestamp int64                  `json:"timestamp"`
	Priority  string                 `json:"priority"`
}

func NewEvent(eventType, source string, payload map[string]interface{}) *Event {
	return &Event{
		ID:        fmt.Sprintf("evt-%d", time.Now().UnixNano()),
		Type:      eventType,
		Source:    source,
		Payload:   payload,
		Timestamp: time.Now().UnixMilli(),
		Priority:  "medium",
	}
}

func (e *Event) ToMap() map[string]interface{} {
	if e.Payload == nil {
		e.Payload = make(map[string]interface{})
	}
	return map[string]interface{}{
		"id":        e.ID,
		"type":      e.Type,
		"source":    e.Source,
		"payload":   e.Payload,
		"timestamp": e.Timestamp,
		"priority":  e.Priority,
	}
}
